--- src/misc.c.org	2006-12-23 00:08:11.000000000 +0900
+++ src/misc.c	2006-12-23 00:08:23.000000000 +0900
@@ -822,6 +822,12 @@
    AddClassToHeap("UnitedLinux");
    }
 
+if (stat("/etc/gentoo-release",&statbuf) != -1)
+   {
+   Verbose("\nThis appears to be a Gentoo Linux system.\n");
+   AddClassToHeap("gentoo");
+   }
+
 if (stat("/etc/vmware",&statbuf) != -1)
    {
    if (S_ISDIR(statbuf.st_mode))
