src/wl/sysだけに発見: .wl_linux.o.d
diff -ru src.orig/wl/sys/wl_linux.c src/wl/sys/wl_linux.c
--- src.orig/wl/sys/wl_linux.c	2009-04-13 01:49:02.000000000 +0900
+++ src/wl/sys/wl_linux.c	2009-04-13 01:50:55.000000000 +0900
@@ -54,8 +54,11 @@
 #define WLC_MAXBSSCFG		1	
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 14)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 30)
 #include <net/ieee80211.h>
 #endif
+#include <net/ieee80211_radiotap.h>
+#endif
 
 #include <asm/system.h>
 #include <asm/io.h>
