From c82e6e787268e7894582222e11c155a088f10964 Mon Sep 17 00:00:00 2001
From: TANABE Ken-ichi <nabeken@tknetworks.org>
Date: Sat, 28 Jun 2008 15:44:00 +0900
Subject: [PATCH] Using config.cert_extension instead of hardcoded cert file extension '.cert'

---
 func/overlord/client.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/func/overlord/client.py b/func/overlord/client.py
index fb436ed..727746b 100755
--- a/func/overlord/client.py
+++ b/func/overlord/client.py
@@ -111,11 +111,11 @@ class Minions(object):
         seperate_gloobs = self.spec.split(";")
         seperate_gloobs = seperate_gloobs + self.new_hosts
         for each_gloob in seperate_gloobs:
-            actual_gloob = "%s/%s.cert" % (self.config.certroot, each_gloob)
+            actual_gloob = "%s/%s.%s" % (self.config.certroot, each_gloob, self.config.cert_extension)
             certs = glob.glob(actual_gloob)
             for cert in certs:
                 self.all_certs.append(cert)
-                host = cert.replace(self.config.certroot,"")[1:-5]
+                host = cert.replace(self.config.certroot,"")[1:-(len(self.config.cert_extension) + 1)]
                 self.all_hosts.append(host)
         return self.all_hosts
 
-- 
1.5.4.5

