commit 4ba08909831d8444978af9559c9f498e2d42ee0f
Author: TANABE Ken-ichi <nabeken@tknetworks.org>
Date:   Thu Oct 8 23:02:59 2009 +0900

    munin-node.conf: Add allow ::1

diff --git a/node/munin-node.conf.in b/node/munin-node.conf.in
index e300ec0..cba31ac 100644
--- a/node/munin-node.conf.in
+++ b/node/munin-node.conf.in
@@ -35,6 +35,7 @@ ignore_file \.pod$
 # may repeat the allow line as many times as you'd like
 
 allow ^127\.0\.0\.1$
+allow ^::1$
 
 # If you have installed the Net::CIDR perl module, you can use
 # multiple cidr_allow and cidr_deny address/mask patterns.  A
