diff -Naur unbound-1.3.3.orig/doc/example.conf.in unbound-1.3.3/doc/example.conf.in
--- unbound-1.3.3.orig/doc/example.conf.in	2009-07-20 23:22:29.000000000 +0900
+++ unbound-1.3.3/doc/example.conf.in	2009-10-05 16:13:27.000000000 +0900
@@ -412,7 +412,7 @@
 remote-control:
 	# Enable remote control with unbound-control(8) here.
 	# set up the keys and certificates with unbound-control-setup.
-	# control-enable: no
+	control-enable: yes
 
 	# what interfaces are listened to for remote control.
 	# give 0.0.0.0 and ::0 to listen to all interfaces.