diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-11-07 00:57:16.000000000 +0900
+++ linden/indra/SConstruct	2007-11-07 00:59:15.000000000 +0900
@@ -111,7 +111,7 @@
 channel = optenv['CHANNEL']
 standalone = optenv['STANDALONE']
 opensource = standalone or optenv['OPENSOURCE']
-enable_fmod = not opensource and optenv['FMOD']
+enable_fmod = optenv['FMOD']
 elfio = optenv['ELFIO']
 
 targets = [ target_param ]
@@ -134,6 +134,7 @@
     'gdk-pixbuf-2.0',
     'glib-2.0',
     'gmodule-2.0',
+    'gthread-2.0',
     'gstreamer-0.10',
     'gtk+-2.0',
     'gtkglext-1.0',
@@ -155,6 +156,9 @@
     'libssl',
     ]
 
+if enable_gstreamer:
+    standalone_pkgs += [ 'gstreamer-0.10' ]
+
 def pkgconfig(opt, pkgs=None):
     if pkgs is None:
         pkgs = standalone_pkgs + standalone_net_pkgs
@@ -611,7 +615,7 @@
         net_external_libs = [ 'curl', 'ssl', 'crypto', 'aprutil-1', 'apr-1' ]
     net_external_libs += [ 'cares', 'expat' ]
 
-    common_external_libs = net_external_libs + [ 'xmlrpc-epi', 'z' ]
+    common_external_libs = net_external_libs + [ 'xmlrpc', 'z' ]
 
     if build_target == 'client':
         if platform == 'linux':
@@ -754,7 +758,7 @@
                             internal_libs + external_libs)
 
         # Rpcserver
-        external_libs = common_external_libs + ['xmlrpc-epi', 'mysqlclient']
+        external_libs = common_external_libs + ['xmlrpc', 'mysqlclient']
         internal_libs = ['llscene', 'llmessage', 'lldatabase', 'llvfs', 
                 'llmath', 'llcommon']
         create_executable('rpcserver/rpcserver' + file_suffix, 'rpcserver',
diff -Naur linden.orig/indra/llimagej2coj/llimagej2coj.cpp linden/indra/llimagej2coj/llimagej2coj.cpp
--- linden.orig/indra/llimagej2coj/llimagej2coj.cpp	2007-09-21 04:33:27.000000000 +0900
+++ linden/indra/llimagej2coj/llimagej2coj.cpp	2007-11-07 00:57:55.000000000 +0900
@@ -31,7 +31,7 @@
 
 // this is defined so that we get static linking.
 #define OPJ_STATIC
-#include "openjpeg/openjpeg.h"
+#include "openjpeg.h"
 
 #include "lltimer.h"
 #include "llmemory.h"
diff -Naur linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp linden/indra/llmessage/llfiltersd2xmlrpc.cpp
--- linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-09-21 04:33:27.000000000 +0900
+++ linden/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-11-07 00:57:55.000000000 +0900
@@ -77,7 +77,7 @@
 
 #include <sstream>
 #include <iterator>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 #include "apr-1/apr_base64.h"
 
 #include "llbuffer.h"
diff -Naur linden.orig/indra/newview/lluserauth.cpp linden/indra/newview/lluserauth.cpp
--- linden.orig/indra/newview/lluserauth.cpp	2007-09-21 04:33:33.000000000 +0900
+++ linden/indra/newview/lluserauth.cpp	2007-11-07 00:57:55.000000000 +0900
@@ -42,7 +42,7 @@
 
 // NOTE: MUST include these after otherincludes since queue gets redefined!?!!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 
 
diff -Naur linden.orig/indra/newview/llvoiceclient.cpp linden/indra/newview/llvoiceclient.cpp
--- linden.orig/indra/newview/llvoiceclient.cpp	2007-09-21 04:33:33.000000000 +0900
+++ linden/indra/newview/llvoiceclient.cpp	2007-11-07 00:57:55.000000000 +0900
@@ -36,7 +36,7 @@
 #include "llvoavatar.h"
 #include "llbufferstream.h"
 #include "llfile.h"
-#include "expat/expat.h"
+#include "expat.h"
 #include "llcallbacklist.h"
 #include "llviewerregion.h"
 #include "llviewernetwork.h"		// for gUserServerChoice
diff -Naur linden.orig/indra/newview/llxmlrpctransaction.cpp linden/indra/newview/llxmlrpctransaction.cpp
--- linden.orig/indra/newview/llxmlrpctransaction.cpp	2007-09-21 04:33:33.000000000 +0900
+++ linden/indra/newview/llxmlrpctransaction.cpp	2007-11-07 00:57:55.000000000 +0900
@@ -34,7 +34,7 @@
 
 // Have to include these last to avoid queue redefinition!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 #include "viewer.h"
 
