diff -Naur linden.orig/indra/newview/llstatusbar.h linden/indra/newview/llstatusbar.h
--- linden.orig/indra/newview/llstatusbar.h	2007-07-07 03:43:29.000000000 +0900
+++ linden/indra/newview/llstatusbar.h	2007-07-09 18:23:13.000000000 +0900
@@ -127,7 +127,7 @@
 	
 	static std::vector<std::string> sDays;
 	static std::vector<std::string> sMonths;
-	static const U32 LLStatusBar::MAX_DATE_STRING_LENGTH;
+	static const U32 MAX_DATE_STRING_LENGTH;
 };
 
 // *HACK: Status bar owns your cached money balance. JC
