diff -Naur linden.orig/indra/newview/viewer.cpp linden/indra/newview/viewer.cpp
--- linden.orig/indra/newview/viewer.cpp	2007-06-26 08:48:31.000000000 +0900
+++ linden/indra/newview/viewer.cpp	2007-07-02 23:57:04.000000000 +0900
@@ -2368,7 +2368,7 @@
 	size_t btpos;
 	for (btpos = 0; btpos < btsize; ++btpos)
 	{
-		fprintf(StraceFile, "%d:\t", btpos);
+		fprintf(StraceFile, "%zd:\t", btpos);
 		int symidx;
 		for (symidx = 0; symidx < nSymNo; ++symidx)
 		{
