diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-03-22 15:34:54.000000000 +0900
+++ linden/indra/SConstruct	2007-03-22 20:46:41.000000000 +0900
@@ -192,6 +192,12 @@
 			include_dirs += [ '../libraries/' + system_str + '/include/atk-1.0' ]
 			include_dirs += [ '../libraries/' + system_str + '/include/ELFIO' ]
 			include_dirs += [ '../libraries/' + system_str + '/include/llfreetype2' ]
+			pipe = os.popen('pkg-config gtk+-2.0 --cflags')
+			flags += pipe.read().rstrip('\n') + ' '
+			pipe.close()
+			pipe = os.popen('freetype-config --cflags')
+			flags += pipe.read().rstrip('\n') + ' '
+			pipe.close()
 
 			# llmozlib stuff
 			if enable_mozlib == 'yes':
@@ -207,7 +213,7 @@
 	### Build type-specific flags ###
 
 	debug_opts = flags + '-fno-inline -O0 -D_DEBUG -DLL_DEBUG=1 '
-	release_opts = flags + '-O2 -DNDEBUG -DLL_RELEASE=1 '
+	release_opts = flags + '-DNDEBUG -DLL_RELEASE=1 '
 	releasenoopt_opts = flags + '-O0 -DNDEBUG -DLL_RELEASE=1 '
 	releasefordownload_opts = flags + '-O2 -DNDEBUG -DLL_RELEASE=1 -DLL_RELEASE_FOR_DOWNLOAD=1 '
 
diff -Naur linden.orig/indra/llcommon/llsdserialize_xml.cpp linden/indra/llcommon/llsdserialize_xml.cpp
--- linden.orig/indra/llcommon/llsdserialize_xml.cpp	2007-03-22 15:34:56.000000000 +0900
+++ linden/indra/llcommon/llsdserialize_xml.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -35,7 +35,7 @@
 
 extern "C"
 {
-#include "expat/expat.h"
+#include "expat.h"
 }
 
 /**
diff -Naur linden.orig/indra/llcommon/llsys.cpp linden/indra/llcommon/llsys.cpp
--- linden.orig/indra/llcommon/llsys.cpp	2007-03-22 15:34:56.000000000 +0900
+++ linden/indra/llcommon/llsys.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -30,7 +30,7 @@
 #include "llsys.h"
 
 #include <iostream>
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include "processor.h"
 
 #if LL_WINDOWS
diff -Naur linden.orig/indra/llimage/llimagejpeg.h linden/indra/llimage/llimagejpeg.h
--- linden.orig/indra/llimage/llimagejpeg.h	2007-03-22 15:34:56.000000000 +0900
+++ linden/indra/llimage/llimagejpeg.h	2007-03-22 19:13:51.000000000 +0900
@@ -33,9 +33,8 @@
 #include "llimage.h"
 
 extern "C" {
-#include "jpeglib/jinclude.h"
-#include "jpeglib/jpeglib.h"
-#include "jpeglib/jerror.h"
+#include "jpeglib.h"
+#include "jerror.h"
 }
 
 class LLImageJPEG : public LLImageFormatted
diff -Naur linden.orig/indra/llimagej2coj/llimagej2coj.cpp linden/indra/llimagej2coj/llimagej2coj.cpp
--- linden.orig/indra/llimagej2coj/llimagej2coj.cpp	2007-03-22 15:34:56.000000000 +0900
+++ linden/indra/llimagej2coj/llimagej2coj.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -30,7 +30,7 @@
 
 // this is defined so that we get static linking.
 #define OPJ_STATIC
-#include "openjpeg/openjpeg.h"
+#include "openjpeg.h"
 
 #include "lltimer.h"
 #include "llmemory.h"
diff -Naur linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp linden/indra/llmessage/llfiltersd2xmlrpc.cpp
--- linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-03-22 15:34:58.000000000 +0900
+++ linden/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -76,7 +76,7 @@
 
 #include <sstream>
 #include <iterator>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 #include "apr-1/apr_base64.h"
 
 #include "llbuffer.h"
diff -Naur linden.orig/indra/llmessage/llhttpassetstorage.cpp linden/indra/llmessage/llhttpassetstorage.cpp
--- linden.orig/indra/llmessage/llhttpassetstorage.cpp	2007-03-22 15:34:58.000000000 +0900
+++ linden/indra/llmessage/llhttpassetstorage.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -36,7 +36,7 @@
 #include "llvfile.h"
 #include "llvfs.h"
 
-#include "zlib/zlib.h"
+#include "zlib.h"
 
 const U32 MAX_RUNNING_REQUESTS = 4;
 const F32 MAX_PROCESSING_TIME = 0.005f;
diff -Naur linden.orig/indra/llrender/llfont.cpp linden/indra/llrender/llfont.cpp
--- linden.orig/indra/llrender/llfont.cpp	2007-03-22 15:34:58.000000000 +0900
+++ linden/indra/llrender/llfont.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -30,11 +30,11 @@
 #include "llfont.h"
 
 // Freetype stuff
-#if LL_LINUX   // I had to do some work to avoid the system-installed FreeType headers... --ryan.
-#include "llfreetype2/freetype/ft2build.h"
-#else
+// #if LL_LINUX   // I had to do some work to avoid the system-installed FreeType headers... --ryan.
+// #include "llfreetype2/freetype/ft2build.h"
+// #else
 #include <ft2build.h>
-#endif
+// #endif
 
 // For some reason, this won't work if it's not wrapped in the ifdef
 #ifdef FT_FREETYPE_H
diff -Naur linden.orig/indra/llxml/llxmlnode.h linden/indra/llxml/llxmlnode.h
--- linden.orig/indra/llxml/llxmlnode.h	2007-03-22 15:35:00.000000000 +0900
+++ linden/indra/llxml/llxmlnode.h	2007-03-22 19:13:51.000000000 +0900
@@ -29,7 +29,7 @@
 #define LL_LLXMLNODE_H
 
 #define XML_STATIC
-#include "expat/expat.h"
+#include "expat.h"
 #include <map>
 
 #include "indra_constants.h"
diff -Naur linden.orig/indra/llxml/llxmlparser.h linden/indra/llxml/llxmlparser.h
--- linden.orig/indra/llxml/llxmlparser.h	2007-03-22 15:35:00.000000000 +0900
+++ linden/indra/llxml/llxmlparser.h	2007-03-22 19:13:51.000000000 +0900
@@ -29,7 +29,7 @@
 #define LL_LLXMLPARSER_H
 
 #define XML_STATIC
-#include "expat/expat.h"
+#include "expat.h"
 
 class LLXmlParser
 {
diff -Naur linden.orig/indra/newview/llfloaterpostcard.cpp linden/indra/newview/llfloaterpostcard.cpp
--- linden.orig/indra/newview/llfloaterpostcard.cpp	2007-03-22 15:35:12.000000000 +0900
+++ linden/indra/newview/llfloaterpostcard.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -237,7 +237,7 @@
 	{	
 	}
 	// *TODO define custom uploadFailed here so it's not such a generic message
-	void LLSendPostcardResponder::uploadComplete(const LLSD& content)
+	void uploadComplete(const LLSD& content)
 	{
 		// we don't care about what the server returns from this post, just clean up the UI
 		LLUploadDialog::modalUploadFinished();
diff -Naur linden.orig/indra/newview/lluserauth.cpp linden/indra/newview/lluserauth.cpp
--- linden.orig/indra/newview/lluserauth.cpp	2007-03-22 15:35:09.000000000 +0900
+++ linden/indra/newview/lluserauth.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -40,7 +40,7 @@
 
 // NOTE: MUST include these after otherincludes since queue gets redefined!?!!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 
 
diff -Naur linden.orig/indra/newview/llviewerobjectlist.cpp linden/indra/newview/llviewerobjectlist.cpp
--- linden.orig/indra/newview/llviewerobjectlist.cpp	2007-03-22 15:35:11.000000000 +0900
+++ linden/indra/newview/llviewerobjectlist.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -60,7 +60,7 @@
 #include "u64.h"
 #include "llviewerimagelist.h"
 #include "lldatapacker.h"
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include "object_flags.h"
 
 extern BOOL gVelocityInterpolate;
diff -Naur linden.orig/indra/newview/llwebbrowserctrl.h linden/indra/newview/llwebbrowserctrl.h
--- linden.orig/indra/newview/llwebbrowserctrl.h	2007-03-22 15:35:05.000000000 +0900
+++ linden/indra/newview/llwebbrowserctrl.h	2007-03-22 19:28:32.000000000 +0900
@@ -91,6 +91,7 @@
 #include "lldynamictexture.h"
 #include "llmozlib.h"
 
+class LLUICtrlFactory;
 class LLViewBorder;
 class LLWebBrowserTexture;
 
@@ -239,7 +240,7 @@
 class LLWebBrowserTexture : public LLDynamicTexture
 {
 	public:
-		LLWebBrowserTexture::LLWebBrowserTexture( S32 width, S32 height, LLWebBrowserCtrl* browserCtrl, int browserWindow );
+		LLWebBrowserTexture( S32 width, S32 height, LLWebBrowserCtrl* browserCtrl, int browserWindow );
 		virtual ~LLWebBrowserTexture();
 
 		virtual void preRender( BOOL clear_depth = TRUE ) {};
diff -Naur linden.orig/indra/newview/llxmlrpctransaction.cpp linden/indra/newview/llxmlrpctransaction.cpp
--- linden.orig/indra/newview/llxmlrpctransaction.cpp	2007-03-22 15:35:10.000000000 +0900
+++ linden/indra/newview/llxmlrpctransaction.cpp	2007-03-22 19:13:51.000000000 +0900
@@ -33,7 +33,7 @@
 
 // Have to include these last to avoid queue redefinition!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 #include "viewer.h"
 
