diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-08-03 06:13:06.000000000 +0900
+++ linden/indra/SConstruct	2007-08-05 08:13:12.000000000 +0900
@@ -158,6 +158,9 @@
     'vorbisfile',
     ]
 
+if enable_gstreamer:
+	standalone_pkgs += [ 'gstreamer-0.10' ]
+
 def pkgconfig(opt, pkgs=None):
     return os.popen('pkg-config %s %s' %
                     (opt, pkgs or ' '.join(standalone_pkgs))).read().strip()
@@ -213,16 +216,7 @@
         """ +
         '../libraries/' + system_str + '/include' )
 
-    if platform == 'linux' and build_target == 'client':
-        if arch == 'x86_64' and os.path.exists('/usr/lib64'):
-            client_external_libs = [File('/usr/lib64/libresolv.a')]
-        else:
-            # Custom libresolv build which avoids a billion flavors of
-            # brokenness prevalent in common libresolvs out there.
-            client_external_libs = ['resolv']
-            include_dirs += ['../libraries/' + system_str + '/include/llresolv8']
-    else:
-        client_external_libs = ['resolv']
+    client_external_libs = ['resolv']
 
     system_link_flags = ''
 
diff -Naur linden.orig/indra/llimagej2coj/llimagej2coj.cpp linden/indra/llimagej2coj/llimagej2coj.cpp
--- linden.orig/indra/llimagej2coj/llimagej2coj.cpp	2007-08-03 06:13:06.000000000 +0900
+++ linden/indra/llimagej2coj/llimagej2coj.cpp	2007-08-04 06:03:28.000000000 +0900
@@ -31,7 +31,7 @@
 
 // this is defined so that we get static linking.
 #define OPJ_STATIC
-#include "openjpeg/openjpeg.h"
+#include "openjpeg.h"
 
 #include "lltimer.h"
 #include "llmemory.h"
diff -Naur linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp linden/indra/llmessage/llfiltersd2xmlrpc.cpp
--- linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-08-03 06:13:07.000000000 +0900
+++ linden/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-08-04 06:03:28.000000000 +0900
@@ -77,7 +77,7 @@
 
 #include <sstream>
 #include <iterator>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 #include "apr-1/apr_base64.h"
 
 #include "llbuffer.h"
diff -Naur linden.orig/indra/newview/llsrv.cpp linden/indra/newview/llsrv.cpp
--- linden.orig/indra/newview/llsrv.cpp	2007-08-03 06:13:14.000000000 +0900
+++ linden/indra/newview/llsrv.cpp	2007-08-04 06:24:20.000000000 +0900
@@ -73,6 +73,7 @@
 
 #include <netdb.h>
 
+#define HOMEGROWN_RESPONSE_PARSER
 #ifdef HOMEGROWN_RESPONSE_PARSER
 
 // We ought to be using libresolv's ns_initparse and ns_parserr to
diff -Naur linden.orig/indra/newview/lluserauth.cpp linden/indra/newview/lluserauth.cpp
--- linden.orig/indra/newview/lluserauth.cpp	2007-08-03 06:13:13.000000000 +0900
+++ linden/indra/newview/lluserauth.cpp	2007-08-04 06:03:28.000000000 +0900
@@ -42,7 +42,7 @@
 
 // NOTE: MUST include these after otherincludes since queue gets redefined!?!!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 
 
diff -Naur linden.orig/indra/newview/llvoiceclient.cpp linden/indra/newview/llvoiceclient.cpp
--- linden.orig/indra/newview/llvoiceclient.cpp	2007-08-03 06:13:14.000000000 +0900
+++ linden/indra/newview/llvoiceclient.cpp	2007-08-05 07:52:19.000000000 +0900
@@ -36,7 +36,7 @@
 #include "llvoavatar.h"
 #include "llbufferstream.h"
 #include "llfile.h"
-#include "expat/expat.h"
+#include "expat.h"
 #include "llcallbacklist.h"
 #include "llviewerregion.h"
 #include "llviewernetwork.h"		// for gUserServerChoice
diff -Naur linden.orig/indra/newview/llwebbrowserctrl.h linden/indra/newview/llwebbrowserctrl.h
--- linden.orig/indra/newview/llwebbrowserctrl.h	2007-08-03 06:13:13.000000000 +0900
+++ linden/indra/newview/llwebbrowserctrl.h	2007-08-04 06:03:28.000000000 +0900
@@ -92,6 +92,7 @@
 #include "lldynamictexture.h"
 #include "llmozlib.h"
 
+class LLUICtrlFactory;
 class LLViewBorder;
 class LLWebBrowserTexture;
 
diff -Naur linden.orig/indra/newview/llxmlrpctransaction.cpp linden/indra/newview/llxmlrpctransaction.cpp
--- linden.orig/indra/newview/llxmlrpctransaction.cpp	2007-08-03 06:13:14.000000000 +0900
+++ linden/indra/newview/llxmlrpctransaction.cpp	2007-08-04 06:03:28.000000000 +0900
@@ -34,7 +34,7 @@
 
 // Have to include these last to avoid queue redefinition!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 #include "viewer.h"
 
