diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-04-26 04:12:56.000000000 +0900
+++ linden/indra/SConstruct	2007-04-26 08:21:20.000000000 +0900
@@ -55,6 +55,7 @@
         allowed_values=('client', 'server', 'all')),
     BoolOption('DISTCC', 'Enabled distcc', True),
     BoolOption('MOZLIB', 'Enabled llmozlib/mozilla support', True),
+    BoolOption('FMOD', 'Enabled fmod support', True),
     BoolOption('COLORGCC', 'Enabled colorgcc', True),
     EnumOption('GRID', 'Client package\'s default grid', 'default',
         allowed_values=('default', 'aditi', 'agni', 'dmz', 'durga', 'firstlook', 'ganga', 'shakti', 'siva', 'soma', 'uma', 'vaak')),
@@ -68,6 +69,7 @@
 target_param = optenv['BTARGET']
 enable_distcc = optenv['DISTCC']
 enable_mozlib = optenv['MOZLIB']
+enable_fmod = optenv['FMOD']
 enable_colorgcc = optenv['COLORGCC']
 grid = optenv['GRID']
 opensource = optenv['OPENSOURCE']
@@ -180,16 +182,22 @@
         flags += '-DLL_LINUX=1 '
         if build_target == 'client':
             flags += '-DAPPID=secondlife -DLL_SDL=1 '
-            if arch == 'x86_64' or arch == 'x86_64cross':
+            if not enable_fmod or arch == 'x86_64' or arch == 'x86_64cross':
                 flags += '-DLL_FMOD=0 '
             flags += '-DLL_X11=1 -DLL_GTK=1 '
-            client_external_libs += [ 'gtk-x11-2.0', 'elfio' ]
+            client_external_libs += [ 'gtk-x11-2.0', 'ELFIO' ]
             include_dirs += [ '../libraries/' + system_str + '/include/gtk-2.0' ]
             include_dirs += [ '../libraries/' + system_str + '/include/glib-2.0']
             include_dirs += [ '../libraries/' + system_str + '/include/pango-1.0' ]
             include_dirs += [ '../libraries/' + system_str + '/include/atk-1.0' ]
             include_dirs += [ '../libraries/' + system_str + '/include/ELFIO' ]
             include_dirs += [ '../libraries/' + system_str + '/include/llfreetype2' ]
+	    pipe = os.popen('pkg-config gtk+-2.0 --cflags')
+	    flags += pipe.read().rstrip('\n') + ' '
+	    pipe.close()
+	    pipe = os.popen('freetype-config --cflags')
+	    flags += pipe.read().rstrip('\n') + ' '
+	    pipe.close()
 
             # llmozlib stuff
             if enable_mozlib:
@@ -415,7 +423,7 @@
 
         external_libs = client_external_libs + common_external_libs + [ 'freetype', 'jpeg', 'SDL', 'GL', 'GLU', 'ogg', 'vorbisenc', 'vorbisfile', 'vorbis', 'db-4.2', 'openjpeg' ]
 
-        if arch != 'x86_64' and arch != 'x86_64cross':
+        if enable_fmod and arch != 'x86_64' and arch != 'x86_64cross':
             external_libs += [ 'fmod-3.75' ]
 
         external_libs.remove('cares')
diff -Naur linden.orig/indra/llcommon/llsdserialize_xml.cpp linden/indra/llcommon/llsdserialize_xml.cpp
--- linden.orig/indra/llcommon/llsdserialize_xml.cpp	2007-04-26 04:12:58.000000000 +0900
+++ linden/indra/llcommon/llsdserialize_xml.cpp	2007-04-26 08:21:20.000000000 +0900
@@ -36,7 +36,7 @@
 
 extern "C"
 {
-#include "expat/expat.h"
+#include "expat.h"
 }
 
 /**
diff -Naur linden.orig/indra/llcommon/llsys.cpp linden/indra/llcommon/llsys.cpp
--- linden.orig/indra/llcommon/llsys.cpp	2007-04-26 04:12:59.000000000 +0900
+++ linden/indra/llcommon/llsys.cpp	2007-04-26 08:21:20.000000000 +0900
@@ -31,7 +31,7 @@
 #include "llsys.h"
 
 #include <iostream>
-#include <zlib/zlib.h>
+#include <zlib.h>
 
 #include "processor.h"
 
diff -Naur linden.orig/indra/llimage/llimagejpeg.h linden/indra/llimage/llimagejpeg.h
--- linden.orig/indra/llimage/llimagejpeg.h	2007-04-26 04:12:59.000000000 +0900
+++ linden/indra/llimage/llimagejpeg.h	2007-04-26 08:21:20.000000000 +0900
@@ -34,8 +34,8 @@
 #include "llimage.h"
 
 extern "C" {
-#include "jpeglib/jpeglib.h"
-#include "jpeglib/jerror.h"
+#include "jpeglib.h"
+#include "jerror.h"
 }
 
 class LLImageJPEG : public LLImageFormatted
diff -Naur linden.orig/indra/llimagej2coj/llimagej2coj.cpp linden/indra/llimagej2coj/llimagej2coj.cpp
--- linden.orig/indra/llimagej2coj/llimagej2coj.cpp	2007-04-26 04:12:59.000000000 +0900
+++ linden/indra/llimagej2coj/llimagej2coj.cpp	2007-04-26 08:21:20.000000000 +0900
@@ -31,7 +31,7 @@
 
 // this is defined so that we get static linking.
 #define OPJ_STATIC
-#include "openjpeg/openjpeg.h"
+#include "openjpeg.h"
 
 #include "lltimer.h"
 #include "llmemory.h"
diff -Naur linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp linden/indra/llmessage/llfiltersd2xmlrpc.cpp
--- linden.orig/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-04-26 04:13:01.000000000 +0900
+++ linden/indra/llmessage/llfiltersd2xmlrpc.cpp	2007-04-26 08:21:20.000000000 +0900
@@ -77,7 +77,7 @@
 
 #include <sstream>
 #include <iterator>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 #include "apr-1/apr_base64.h"
 
 #include "llbuffer.h"
diff -Naur linden.orig/indra/llmessage/llhttpassetstorage.cpp linden/indra/llmessage/llhttpassetstorage.cpp
--- linden.orig/indra/llmessage/llhttpassetstorage.cpp	2007-04-26 04:13:01.000000000 +0900
+++ linden/indra/llmessage/llhttpassetstorage.cpp	2007-04-26 08:21:20.000000000 +0900
@@ -37,7 +37,7 @@
 #include "llvfile.h"
 #include "llvfs.h"
 
-#include "zlib/zlib.h"
+#include "zlib.h"
 
 const U32 MAX_RUNNING_REQUESTS = 1;
 const F32 MAX_PROCESSING_TIME = 0.005f;
diff -Naur linden.orig/indra/llrender/llfont.cpp linden/indra/llrender/llfont.cpp
--- linden.orig/indra/llrender/llfont.cpp	2007-04-26 04:13:02.000000000 +0900
+++ linden/indra/llrender/llfont.cpp	2007-04-26 08:21:20.000000000 +0900
@@ -31,11 +31,11 @@
 #include "llfont.h"
 
 // Freetype stuff
-#if LL_LINUX   // I had to do some work to avoid the system-installed FreeType headers... --ryan.
-#include "llfreetype2/freetype/ft2build.h"
-#else
+// #if LL_LINUX   // I had to do some work to avoid the system-installed FreeType headers... --ryan.
+// #include "llfreetype2/freetype/ft2build.h"
+// #else
 #include <ft2build.h>
-#endif
+// #endif
 
 // For some reason, this won't work if it's not wrapped in the ifdef
 #ifdef FT_FREETYPE_H
diff -Naur linden.orig/indra/llxml/llxmlnode.h linden/indra/llxml/llxmlnode.h
--- linden.orig/indra/llxml/llxmlnode.h	2007-04-26 04:13:05.000000000 +0900
+++ linden/indra/llxml/llxmlnode.h	2007-04-26 08:21:20.000000000 +0900
@@ -30,7 +30,7 @@
 #define LL_LLXMLNODE_H
 
 #define XML_STATIC
-#include "expat/expat.h"
+#include "expat.h"
 #include <map>
 
 #include "indra_constants.h"
diff -Naur linden.orig/indra/llxml/llxmlparser.h linden/indra/llxml/llxmlparser.h
--- linden.orig/indra/llxml/llxmlparser.h	2007-04-26 04:13:05.000000000 +0900
+++ linden/indra/llxml/llxmlparser.h	2007-04-26 08:21:20.000000000 +0900
@@ -30,7 +30,7 @@
 #define LL_LLXMLPARSER_H
 
 #define XML_STATIC
-#include "expat/expat.h"
+#include "expat.h"
 
 class LLXmlParser
 {
diff -Naur linden.orig/indra/newview/lluserauth.cpp linden/indra/newview/lluserauth.cpp
--- linden.orig/indra/newview/lluserauth.cpp	2007-04-26 04:13:13.000000000 +0900
+++ linden/indra/newview/lluserauth.cpp	2007-04-26 08:21:42.000000000 +0900
@@ -42,7 +42,7 @@
 
 // NOTE: MUST include these after otherincludes since queue gets redefined!?!!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 
 
diff -Naur linden.orig/indra/newview/llviewerobjectlist.cpp linden/indra/newview/llviewerobjectlist.cpp
--- linden.orig/indra/newview/llviewerobjectlist.cpp	2007-04-26 04:13:15.000000000 +0900
+++ linden/indra/newview/llviewerobjectlist.cpp	2007-04-26 08:21:42.000000000 +0900
@@ -61,7 +61,7 @@
 #include "u64.h"
 #include "llviewerimagelist.h"
 #include "lldatapacker.h"
-#include <zlib/zlib.h>
+#include <zlib.h>
 #include "object_flags.h"
 
 extern BOOL gVelocityInterpolate;
diff -Naur linden.orig/indra/newview/llwebbrowserctrl.h linden/indra/newview/llwebbrowserctrl.h
--- linden.orig/indra/newview/llwebbrowserctrl.h	2007-04-26 04:13:10.000000000 +0900
+++ linden/indra/newview/llwebbrowserctrl.h	2007-04-26 08:21:42.000000000 +0900
@@ -92,6 +92,7 @@
 #include "lldynamictexture.h"
 #include "llmozlib.h"
 
+class LLUICtrlFactory;
 class LLViewBorder;
 class LLWebBrowserTexture;
 
@@ -249,7 +250,7 @@
 class LLWebBrowserTexture : public LLDynamicTexture
 {
 	public:
-		LLWebBrowserTexture::LLWebBrowserTexture( S32 width, S32 height, LLWebBrowserCtrl* browserCtrl, int browserWindow );
+		LLWebBrowserTexture( S32 width, S32 height, LLWebBrowserCtrl* browserCtrl, int browserWindow );
 		virtual ~LLWebBrowserTexture();
 
 		virtual void preRender( BOOL clear_depth = TRUE ) {};
diff -Naur linden.orig/indra/newview/llxmlrpctransaction.cpp linden/indra/newview/llxmlrpctransaction.cpp
--- linden.orig/indra/newview/llxmlrpctransaction.cpp	2007-04-26 04:13:14.000000000 +0900
+++ linden/indra/newview/llxmlrpctransaction.cpp	2007-04-26 08:21:42.000000000 +0900
@@ -34,7 +34,7 @@
 
 // Have to include these last to avoid queue redefinition!
 #include <curl/curl.h>
-#include <xmlrpc-epi/xmlrpc.h>
+#include <xmlrpc.h>
 
 #include "viewer.h"
 
