diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2008-04-08 11:37:53.000000000 +0900
+++ linden/indra/SConstruct	2008-05-06 00:56:08.000000000 +0900
@@ -118,7 +118,7 @@
 standalone = optenv['STANDALONE']
 runtests = optenv['RUNTESTS']
 opensource = standalone or optenv['OPENSOURCE']
-enable_fmod = not opensource and optenv['FMOD']
+enable_fmod = optenv['FMOD']
 elfio = optenv['ELFIO']
 
 targets = [ target_param ]
@@ -141,6 +141,7 @@
     'gdk-pixbuf-2.0',
     'glib-2.0',
     'gmodule-2.0',
+    'gthread-2.0',
     'gtk+-2.0',
     'libpng',
     'pango',
@@ -161,6 +162,9 @@
     'libssl',
     ]
 
+if enable_gstreamer:
+    standalone_pkgs += [ 'gstreamer-0.10' ]
+
 def pkgconfig(opt, pkgs=None):
     if pkgs is None:
         pkgs = standalone_pkgs + standalone_net_pkgs
@@ -249,7 +253,7 @@
     ##############
 
     # Generic GCC flags
-    cflags = '-g -pipe -Wall -Wno-reorder -Wno-trigraphs -Wno-sign-compare -Werror -fexceptions '
+    cflags = '-g -pipe -Wall -Wno-reorder -Wno-trigraphs -Wno-sign-compare -fexceptions '
     cxxflags = ''
     cppflags = '-D_FORTIFY_SOURCE=2 '
     if standalone:
diff -Naur linden.orig/indra/llimagej2coj/llimagej2coj.cpp linden/indra/llimagej2coj/llimagej2coj.cpp
--- linden.orig/indra/llimagej2coj/llimagej2coj.cpp	2008-04-08 11:37:55.000000000 +0900
+++ linden/indra/llimagej2coj/llimagej2coj.cpp	2008-05-06 00:56:08.000000000 +0900
@@ -33,7 +33,7 @@
 #include "llimagej2coj.h"
 
 // this is defined so that we get static linking.
-#include "openjpeg/openjpeg.h"
+#include "openjpeg.h"
 
 #include "lltimer.h"
 #include "llmemory.h"
diff -Naur linden.orig/indra/llmath/llquaternion.cpp linden/indra/llmath/llquaternion.cpp
--- linden.orig/indra/llmath/llquaternion.cpp	2008-04-08 11:37:55.000000000 +0900
+++ linden/indra/llmath/llquaternion.cpp	2008-05-06 00:56:08.000000000 +0900
@@ -698,22 +698,22 @@
 	{
 	default:
 	case LLQuaternion::XYZ:
-		p = "XYZ";
+		strcpy(p, "XYZ");
 		break;
 	case LLQuaternion::YZX:
-		p = "YZX";
+		strcpy(p, "YZX");
 		break;
 	case LLQuaternion::ZXY:
-		p = "ZXY";
+		strcpy(p, "ZXY");
 		break;
 	case LLQuaternion::XZY:
-		p = "XZY";
+		strcpy(p, "XZY");
 		break;
 	case LLQuaternion::YXZ:
-		p = "YXZ";
+		strcpy(p, "YXZ");
 		break;
 	case LLQuaternion::ZYX:
-		p = "ZYX";
+		strcpy(p, "ZYX");
 		break;
 	}
 	return p;
diff -Naur linden.orig/indra/newview/llvoiceclient.cpp linden/indra/newview/llvoiceclient.cpp
--- linden.orig/indra/newview/llvoiceclient.cpp	2008-04-08 11:38:06.000000000 +0900
+++ linden/indra/newview/llvoiceclient.cpp	2008-05-06 00:56:08.000000000 +0900
@@ -39,7 +39,7 @@
 #include "llvoavatar.h"
 #include "llbufferstream.h"
 #include "llfile.h"
-#include "expat/expat.h"
+#include "expat.h"
 #include "llcallbacklist.h"
 #include "llviewerregion.h"
 #include "llviewernetwork.h"		// for gGridChoice
